Applications/spire/include/spire/login/login_window.hpp

```diff
@@ -1,8 +1,19 @@
 #ifndef SPIRE_LOGIN_WINDOW_HPP
 #define SPIRE_LOGIN_WINDOW_HPP
+#include <string>
 #include <boost/signals2/connection.hpp>
+#include <QEvent>
+#include <QKeyEvent>
+#include <QLabel>
+#include <QLineEdit>
+#include <QMouseEvent>
+#include <QPoint>
+#include <QPushButton>
 #include <QWidget>
 #include "spire/login/login.hpp"
+#include "spire/login/chroma_hash_widget.hpp"
```

Comment:
Don't need to include chroma_hash_widget, flat_button, or icon_button in the header file. Include them instead in the source file.